app/query/[id]/page.tsx

```diff
@@ -0,0 +1,43 @@
+import QueryResult from "./QueryResult";
+import { Metadata } from "next";
+
+
+type MetaProps = {
+  params: { id: string }
+}
+
+
+export async function generateMetadata(
+    { params }: MetaProps
+): Promise<Metadata> {
+  // read route params
+  const id = params.id
+
+  // we want to get the data then can pass in as props
+
+  const res = await fetch(`https://www.asksalesforce.ai/api/query/${id}`, {
+    method: "GET",
+  });
+  const { code, userContent, result } = await res.json();
+
+  return {
+    openGraph: {
+      title: "Salesforce SOQL Generator",
```

Comment:
Do you need to override the title and description?

Reply:
not sure